pykafka/cli/kafka_tools.py

```diff
@@ -54,11 +55,16 @@ def fetch_consumer_lag(client, topic, consumer_group):
     """
     latest_offsets = fetch_offsets(client, topic, 'latest')
     consumer = topic.get_simple_consumer(consumer_group=consumer_group,
-                                         auto_start=False, 
+                                         auto_start=False,
                                          reset_offset_on_fetch=False)
     current_offsets = consumer.fetch_offsets()
-    return {p_id: (latest_offsets[p_id].offset[0], res.offset)
-            for p_id, res in current_offsets}
+    consumer_id_dict = {}
+    for p_id, stat in current_offsets:
+        info = ast.literal_eval(stat[1])
+        consumer_id_dict[p_id] = [info['consumer_id'], info['hostname']]
+    return {p_id: (latest_offsets[p_id].offset[0], res.offset,
+                   consumer_id_dict[p_id][0],
+                   consumer_id_dict[p_id][1], ) for p_id, res in current_offsets}
```

Comment:
This iteration over `current_offsets` is redundant with the one you've added above. Only one pass over `current_offsets` is necessary to retrieve the relevant info, so there should only be one `for` in this logic.

Reply:
Hi, I did some changes to make it only one for loop